Alloy/Renderers/PointsRenderer.swift

```diff
@@ -0,0 +1,143 @@
+//
+//  PointsRenderer.swift
+//  Alloy
+//
+//  Created by Eugene Bokhan on 26/04/2019.
+//
+
+import Metal
+import simd
+
+@available(iOS 11.3, tvOS 11.3, macOS 10.13, *)
+final public class PointsRenderer {
+
+    public enum Errors: Error {
+        case functionCreationFailed
+        case libraryCreationFailed
+        case wrongRenderTargetTextureUsage
+        case missingRenderTarget
+    }
+
+    // MARK: - Properties
+
+    /// Point positions described in a normalized coodrinate system.
+    public var pointsPositions: [simd_float2] {
+        set {
+            var pointsPositions = newValue
+            self.pointCount = pointsPositions.count
+            self.pointsPositionsBuffer = self.context.device
+                .makeBuffer(bytes: &pointsPositions,
+                            length: MemoryLayout<simd_float2>.stride * pointsPositions.count,
+                            options: .storageModeShared)
+        }
+        get {
+            if let pointsPositionsBuffer = self.pointsPositionsBuffer,
+                let pointsPositions = pointsPositionsBuffer
+                    .array(of: simd_float2.self,
+                           count: pointsPositionsBuffer.length / MemoryLayout<simd_float2>.stride) {
+                return pointsPositions
+            } else {
+                return []
+            }
+        }
+    }
+    /// Point color. Red in default.
+    public var color: vector_float4 = .init(1, 0, 0, 1)
+    /// Point size in pixels
+    public var pointSize: Float = 40
+
+    private var pointsPositionsBuffer: MTLBuffer?
+    private var pointCount: Int = 0
+
+    private let context: MTLContext
```

Comment:
it definetely should be public 

we can also think of stroing/injecting MTLLibrary instead